src/remark/variablesPlugin.js

```diff
@@ -1,51 +1,64 @@
-const path = require("path");
-const visit = require("unist-util-visit");
+const path = require("path")
+const visit = require("unist-util-visit")
 
-const variables = require(path.resolve("variables"));
-const objProp = require(path.resolve("src/utils/objProp"));
-
-const pathPrefixes = {
-	cloud: "/docs/calico-cloud/",
-	enterprise: "/docs/calico-enterprise/",
-	openSource: "/docs/calico/",
-};
+const variables = require(path.resolve("variables"))
+const objProp = require(path.resolve("src/utils/objProp"))
+const varRegex = RegExp(/\{\{[ \t]*([\w.]+)[ \t]*}}/, 'g')
 
 function variablesPlugin(_options) {
 	async function transformer(tree, file) {
-		const posixFriendlyPath = convertToPosixFriendlyPath(file.path);
-		const productVariables = posixFriendlyPath.includes(pathPrefixes.cloud)
-			? variables.cloud
-			: posixFriendlyPath.includes(pathPrefixes.enterprise)
-				? variables.enterprise
-				: posixFriendlyPath.includes(pathPrefixes.openSource)
-					? variables.openSource
-					: null;
-
-		if (!productVariables) {
-			return;
-		}
+		const contextVariables = getContextVariables(file, variables)
+		visit(tree, () => true, (node) => {
+			for (let prop in node) {
+				if (!Object.prototype.hasOwnProperty.call(node, prop)) continue
+				if (prop === 'type' || typeof node[prop] !== 'string') continue
 
-		visit(tree, ["text", "jsx", "code", "link", "inlineCode"], (node) => {
-			const valueKey = node.type === "link" ? "url" : "value";
+				node[prop] = node[prop].replaceAll(varRegex, (match, varName) => {
+						let varValue
+						for (let cv of contextVariables) {
+							varValue = objProp(cv, varName)
+							if (varValue) return String(varValue)
+						}
+						varValue = objProp(variables.global, varName)
+						if (varValue) return String(varValue)
+						return match
+					},
+				)
+			}
+		})
+	}
+
+	return transformer
+}
 
-			node[valueKey] = node[valueKey].replaceAll(
-				/{{([\w.]+)}}/g,
-				(match, varName) => {
-					const varValue = objProp(productVariables, varName);
-					if (!varValue) {
-						return match;
+function getContextVariables(file, variables) {
+	let cvars = []
+	const dpName = 'docsPathPrefix'
+	const posixFriendlyPath = convertToPosixFriendlyPath(file.path)
+
+	for (let p in variables) {
+		if (!Object.prototype.hasOwnProperty.call(variables, p)) continue
```

Comment:
Do we really need this check? `for ... in` should iterate only on the existing properties of an object.

Reply:
Normally, yes. But, this is safer as I understand things. There can be times when you will iterate through a property that is not a normal property. Perhaps the world as moved on since I was last coding javascript though! 😄